terraform/scripts/provision.sh

```diff
@@ -4,10 +4,20 @@
 # Use it to bootstrap the host.
 
 # drue's ssh key
-echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDORl4HnMm6j9OsjXENsmJM73OFqBsNCWGmQ0Lucir7Wh+AxCRxDxXmtpnSmW1ZO3Foo0xVC0ie0BYCg3HkwFqst/0Ho7q0yK5zqnZ+oFIgyYdCpnsdNCOgdMTns9NRkW8PG7QfCsLH46h5JbxYp+MRysklPSO4l1sKXf8bTYxowTctC9qIFR5847Djq7xNpK3gJN2Z2nPu9mvjKgK5pwhOuP9dk3aLvUW0sInKn6tYkvsGKRnSv1CbiUesfBDgmwyLSQ8GgKF575jgaI11gTkJXVHxmb6P0pSUgnXBXRggz+qiJBCrqkUiLhwn6uPCZ8nXWShfTqwg+682qda/3mKX drue" >> .ssh/authorized_keys
+echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDORl4HnMm6j9OsjXENsmJM73OFqBsNCWGmQ0Lucir7Wh+AxCRxDxXmtpnSmW1ZO3Foo0xVC0ie0BYCg3HkwFqst/0Ho7q0yK5zqnZ+oFIgyYdCpnsdNCOgdMTns9NRkW8PG7QfCsLH46h5JbxYp+MRysklPSO4l1sKXf8bTYxowTctC9qIFR5847Djq7xNpK3gJN2Z2nPu9mvjKgK5pwhOuP9dk3aLvUW0sInKn6tYkvsGKRnSv1CbiUesfBDgmwyLSQ8GgKF575jgaI11gTkJXVHxmb6P0pSUgnXBXRggz+qiJBCrqkUiLhwn6uPCZ8nXWShfTqwg+682qda/3mKX drue" >> /home/ubuntu/.ssh/authorized_keys
 
 # terceiro's ssh key
-echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC4XbnFOoWpbwEiX0k6YsJQteanZft5E8IuzZK9JjZ48GC+4fU8cpFxc8Na5MSkxGawmpATwCaMvrGZyP1hKqxAgnTO8qAkBIhn7HvGdMJ870ePjOCPNg9OiRUtWV85tNDsPsq1rjMPePMMIg3jZeVXhHjBy1lytOLp3L48qMBHOlUzJzBmZ8VDzREY+Xo/vEx+Q15EB9/qPsbYvruyIQOiDKWxasIcKQNQvoaa082U1ZEb7OyNum81E93waccWENNWhovVJ03yyfKJ/i9ToS6kcm5DfEFSoa0yPSpClJ3B2daVIBJxyX1NXZilZhYvKHVEjamLsekLI5fgMG2Nk51D terceiro" >> .ssh/authorized_keys
+echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC4XbnFOoWpbwEiX0k6YsJQteanZft5E8IuzZK9JjZ48GC+4fU8cpFxc8Na5MSkxGawmpATwCaMvrGZyP1hKqxAgnTO8qAkBIhn7HvGdMJ870ePjOCPNg9OiRUtWV85tNDsPsq1rjMPePMMIg3jZeVXhHjBy1lytOLp3L48qMBHOlUzJzBmZ8VDzREY+Xo/vEx+Q15EB9/qPsbYvruyIQOiDKWxasIcKQNQvoaa082U1ZEb7OyNum81E93waccWENNWhovVJ03yyfKJ/i9ToS6kcm5DfEFSoa0yPSpClJ3B2daVIBJxyX1NXZilZhYvKHVEjamLsekLI5fgMG2Nk51D terceiro" >> /home/ubuntu/.ssh/authorized_keys
 
 # mwasilew's ssh key
-echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDLg0a071B8RSBVYxXoGa/dMZ7jaiLdEM8tmMIg0VTFAh8xPOUIJ8HWurQdc+mq6mMtBFqKGJ5YshLEXK//CKRW+lR+2eTEpjMLfoOR7u1zxU35+lAxbVavOwEHzjaPaGypmaqwWvNdlgsg2gl5Qo7B2f9nEnHtieAW7qI/1agjorB8/I12H2H2iC7GWKptRq1wPRp2sgwq2Bk286xTOESFV+iv0tzT5GepJUexXmF69xqlkW5uznA7LV8DqPQk/5n42K8i5gMjH+ulEDTc1/aMVjjTaSIEbEsvyvhXCXa7PpCRdXT/vodKHnRUwJPu5lkX5m9WSpl6E0RdqQY/3WTn mwasilew" >> .ssh/authorized_keys
+echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDLg0a071B8RSBVYxXoGa/dMZ7jaiLdEM8tmMIg0VTFAh8xPOUIJ8HWurQdc+mq6mMtBFqKGJ5YshLEXK//CKRW+lR+2eTEpjMLfoOR7u1zxU35+lAxbVavOwEHzjaPaGypmaqwWvNdlgsg2gl5Qo7B2f9nEnHtieAW7qI/1agjorB8/I12H2H2iC7GWKptRq1wPRp2sgwq2Bk286xTOESFV+iv0tzT5GepJUexXmF69xqlkW5uznA7LV8DqPQk/5n42K8i5gMjH+ulEDTc1/aMVjjTaSIEbEsvyvhXCXa7PpCRdXT/vodKHnRUwJPu5lkX5m9WSpl6E0RdqQY/3WTn mwasilew" >> /home/ubuntu/.ssh/authorized_keys
+
+retry() {
+    n=10
+    while [ "$n" -gt 0 ] && ! "$@"; do
+        sleep 5s
```

Comment:
i would do something like 5*n here to give it a little backoff

Reply:
ack